infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -526,6 +567,130 @@ private void analyzePollReportAndTriggerHandler(PollReport pollReport) {
     }
 
     /**
+     * Corrects out of phase epochs by resealing the servers.
+     * This would also need to update trailing layout servers.
+     *
+     * @param pollReport Poll Report from running the failure detection policy.
+     */
+    private void correctOutOfPhaseEpochs(PollReport pollReport) {
+        final Map<String, Long> outOfPhaseEpochNodes = pollReport.getOutOfPhaseEpochNodes();
+        if (outOfPhaseEpochNodes.isEmpty()) {
+            return;
+        }
+
+        try {
+            // Query all layout servers to get quorum Layout.
+            Map<String, CompletableFuture<Layout>> layoutCompletableFutureMap = new HashMap<>();
```

Comment:
make this  a util method

Reply:
I require this map in `updateTrailingLayoutServers` and cannot abstract this to a util method.
I cannot use the `outOfPhaseEpochs` list as there could be a scenario where the layout server was sealed successfully but did not receive the commit message in paxos. This will not trigger a `WrongEpochException` but the layout server still has to be updated.

Regardless, we need a util method to fetch the quorum layout.